testing/trino-benchto-benchmarks/src/test/java/io/trino/sql/planner/statistics/GlueStatisticsGenerator.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.statistics;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.hive.HiveColumnHandle;
+import io.trino.plugin.hive.HiveConfig;
+import io.trino.plugin.hive.HiveConnector;
+import io.trino.plugin.hive.HiveMetadata;
+import io.trino.plugin.hive.HivePartition;
+import io.trino.plugin.hive.HivePartitionManager;
+import io.trino.plugin.hive.HivePartitionResult;
+import io.trino.plugin.hive.HivePlugin;
+import io.trino.plugin.hive.HiveTableHandle;
+import io.trino.plugin.hive.PartitionStatistics;
+import io.trino.spi.Plugin;
+import io.trino.spi.connector.ConnectorFactory;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.Constraint;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.testing.TestingConnectorContext;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static com.google.common.collect.MoreCollectors.toOptional;
+import static com.google.common.collect.Streams.stream;
+import static io.trino.plugin.hive.HivePartition.UNPARTITIONED_ID;
+import static io.trino.plugin.hive.HiveTestUtils.SESSION;
+import static io.trino.plugin.hive.statistics.MetastoreHiveStatisticsProvider.getPartitionsStatistics;
+import static io.trino.spi.transaction.IsolationLevel.READ_UNCOMMITTED;
+import static io.trino.sql.planner.TestHiveTpcdsCostBasedPlan.TPCDS_SF1000_STATISTICS_REPOSITORY;
+import static io.trino.sql.planner.TestHiveTpchCostBasedPlan.TPCH_SF1000_STATISTICS_REPOSITORY;
+import static java.util.Objects.requireNonNull;
+
+public class GlueStatisticsGenerator
+        implements AutoCloseable
+{
+    private final HiveConnector connector;
+    private final ConnectorTransactionHandle transaction;
+    private final HiveMetadata metadata;
+    private final HivePartitionManager partitionManager;
+
+    public GlueStatisticsGenerator()
```

Comment:
I think `SHOW STATS FOR (SELECT * FROM TABLE WHERE column = partitionName)` gives the the stats for a partition. Can we take the result of that query, use it to populate `PartitionStatistics` and dump that to file ?
We would need to do conversion of engine stats to hive stats, but I think we do that when we store the stats collected for hive through ANALYZE as well (see `createHiveColumnStatistics`).
This would be a bit cumbersome but it would reduce the low level dependencies that we have in the current approach.

Reply:
Actually the above way isn't ideal either. E.g. if somebody adds column histogram stats at partition level to hive then we won't be able to derive that from SHOW STATS output.
So it looks like we have to implement this logic at hive connector level.
Could we try providing a HiveMetastore implementation to TestingHiveConnectorFactory which wraps the getTableStatistics method of actual metastore implementation and dumps stats to file before returning them to engine ? We could then run SHOW STATS queries to make it dump stats on the required partitions and tables.

Reply:
Also check if we can setup RecordingHiveMetastore using `hive.metastore-recording-path`and make that dump hive stats for us by running SHOW STATS queries